doc/source/guide/data_types/timeseries.rst

```diff
@@ -0,0 +1,378 @@
+`~sunpy.data.sample.EVE_LIGHTCURVE` and `~sunpy.data.sample.GOES_LIGHTCURVE`.
+To create the sample `sunpy.timeseries.sources.goes.GOESLightCurve` type the
+ following into your interactive Python shell: ::
+
+    >>> import sunpy
+    >>> import sunpy.timeseries
+    >>> import sunpy.data.sample
+    >>> my_timeseries = sunpy.timeseries.TimeSeries(sunpy.data.sample.GOES_LIGHTCURVE, source='GOES')
+
+This is calling the TimeSeries factory (TimeSeries) to create a time series from a sample fits file.
+If you have not downloaded the data already you should get an error and some
+instruction on how to download the sample data.
+
+The variable my_timeseries is a :ref:`timeseries` object. To create one from a
+local GOES FITS file try the following: ::
+
+    >>> my_timeseries = sunpy.timeseries.TimeSeries('/mydirectory/myts.fits', source='GOES')   # doctest: +SKIP
+
+SunPy can automatically detect the source for most FITS files. However timeseries
+(and lightcurve) data are stored in a variety of file formats (FITS, txt, csv)
+and it's not always possible to detect the source. For this reason, it's good
+practice to explicitly state the source for the file.
+The factory has the ability to make a list of TimeSeries objects using a list of filepaths, a folder or a glob, for example: ::
+
+    >>> my_ts_list = sunpy.timeseries.TimeSeries('filepath1', 'filepath2', source='GOES')   # doctest: +SKIP
+    >>> my_ts_list = sunpy.timeseries.TimeSeries('/goesdirectory/', source='GOES')   # doctest: +SKIP
+    >>> my_ts_list = sunpy.timeseries.TimeSeries(glob, source='GOES')   # doctest: +SKIP
+
+Note that the factory will only work with files from a single source, generating a source specific child of the `~sunpy.timeseries.timeseriesbase.GenericTimeSeries` class such as the `~sunpy.timeseries.sources.goes.GOESLightCurve` above. For this reason all the files should be from that same source for the factory to work correctly.
+
+1.1 Creating a Single TimeSeries from Multiple Files
+----------------------------------------------------
+
+You can create a single time series from multiple files for a given source using
+the keyword argument concatenate=True, such as:
+
+    >>> my_timeseries = sunpy.timeseries.TimeSeries('/mydirectory/myts1.fits', '/mydirectory/myts2.fits', source='GOES', concatenate=True)   # doctest: +SKIP
+
+Note these must all be from the same source/instrument if using concatenate from within the TimeSeries factory.
+The time series concatenate method can be used to make a time series from multiple TimeSeries from different sources once they are already in the form of a TimeSeries objects.
+
+2. Creating Custom TimeSeries
+-----------------------------
+
+Sometimes you will have data that you want to create into a TimeSeries. You can use the factory to create a `~sunpy.timeseries.timeseriesbase.GenericTimeSeries` from a variety of data sources currently including `pandas.DataFrame` and `astropy.table.table.Table`.
```

Comment:
I think you could technically try to create a instrument source file from data manually, but unless your data is correctly formatted (has the correct columns) it'd mess up on the peek method.
Generally if you use your own data it would be a generic TS, is this logical?